web/src/engine/websites/NetComics.ts

```diff
@@ -0,0 +1,119 @@
+            { key: 'EN', label: R.Tags_Language_English }, //TODO : create langage string without flag emoji or support emoji in dialog?
+            { key: 'ES', label: R.Tags_Language_Spanish },
+            { key: 'FR', label: R.Tags_Language_French },
+            { key: 'DE', label: R.Tags_Language_German },
+        );
+
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return /https:\/\/www\.netcomics\.com\/[a-z]{2}\/comic\/[^/]+/.test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const site = url.match(/\/([a-z]{2})\/comic\//)[1].toUpperCase();
+        const slug = url.match(/\/comic\/(\S+)/)[1];
+        const request = new Request(new URL(`/api/v1/title/comic/${slug}/${site}`, this.apiUrl), {
+            headers: this.getRequestHeaders(site)
+        });
+        const { data } = await FetchJSON<APIResult<APIManga>>(request);
+        return new Manga(this, provider, data.title_id.toString(), data.title_name.trim());
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList : Manga[]= [];
+        for (let page = 1, run = true; run; page++) {
+            const mangas = await this.getMangasFromPage(page, provider);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList.distinct();
+    }
+
+    async getMangasFromPage(page: number, provider: MangaPlugin) {
+        const request = new Request(new URL(`/api/v1/title/genre?no=${page}&size=18&genre=`, this.apiUrl), {
+            headers: this.getRequestHeaders(this.Settings.language.Value as string)
+        });
+        const { data } = await FetchJSON<APIResult<APIManga[]>>(request);
+        return data.map(manga => new Manga(this, provider, manga.title_id.toString(), manga.title_name.trim()));
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const request = new Request(new URL(`/api/v1/chapter/order/${manga.Identifier}/rent`, this.apiUrl));
+        const { data } = await FetchJSON <APIResult<APIChapter[]>>(request);
+        return data.map(chapter => {
+            const title = [chapter.chapter_no.toString(), chapter.chapter_name.trim()].join(' ').trim();
+            return new Chapter(this, manga, chapter.chapter_id.toString(), title);
+        });
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        try {
+            const request = new Request(new URL(`/api/v1/chapter/viewer/625/${chapter.Parent.Identifier}/${chapter.Identifier}?otp=`, this.apiUrl));
+            const { data } = await FetchJSON<APIResult<APIPages>>(request);
+            return data.images.map(image => new Page(this, chapter, new URL(image.image_url)));
+        } catch (e) {
+            throw Error(W.Plugin_Common_Chapter_UnavailableError);
```

Comment:
This will catch all errors (e.g., Out of Memory, JSON Parse, ...) with the same message, is this by intention?
Would be more appropriate to check the exact conditions for the chapter being unavailable (e.g., http response 403).